tempesta_fw/cache.c

```diff
@@ -611,10 +666,10 @@ tfw_cache_process(TfwHttpReq *req, TfwHttpResp *resp,
 	TfwWorkTasklet *ct;
 	TfwCWork cw;
 
-	if (!cache_cfg.cache || !tfw_cache_msg_cacheable(req)) {
-		action(req, resp);
-		return 0;
-	}
+	if (!cache_cfg.cache || !tfw_cache_msg_cacheable(req))
+		goto dont_cache;
+	if (!tfw_cache_employ(req, resp))
```

Comment:
Yes, this is the only place where we should check `req` and `resp` for NULL because we don't know what we're processing, request or response.